### PR TITLE
added llama3 prompt

### PR DESCRIPTION
**Pull request by hirschrobert**
I was picking up issue #1889 , but had to change it a bit. meta documentation is here https://llama.meta.com/docs/model-cards-and-prompt-formats/meta-llama-3/.
Also some issues with double BOS tokens, so no BOS tokens here, see https://github.com/abetlen/llama-cpp-python/pull/1439.

### CODE REVIEW

File: private_gpt/components/llm/prompt_helper.py
Comment (jaluma) [inline private_gpt/components/llm/prompt_helper.py:1]: It would be great to have a unit test to test the output as well as the rest of the `PromptStyles`.

**Comment by jaluma** [comment]
> Hey @imartinez , is it possible to change the pull request rules so that workflows are run without approval. What is the reason for it? Without approval, tests can be run already and code improved by contributors beforehand so that merging by a maintainer should only be a formal issue. This should also speed up things. Many thanks!

We have just changed how workflows are executed. Hopefully there won't be any more problems in the future :)

File: pyproject.toml
Comment (jaluma) [inline pyproject.toml:1]: This change remove a warning calling `lint`...

**Comment by jaluma** [comment]
Updated PR with requested changes!

**Review by imartinez — APPROVED** [review]
👏 🦙

**Review by jaluma — APPROVED** [review]

**Pull request merged**